administrator/components/com_fields/models/fields/type.php

```diff
@@ -91,4 +107,31 @@ protected function getOptions ()
 
 		return $options;
 	}
+
+	/**
+	 * Parses the file with the given path. If it is a class starting with the
+	 * name JFormField and implementing JFormDomfieldinterface, then the class name is returned.
+	 *
+	 * @param string $path
+	 * @return string|boolean
+	 */
+	private function getClassNameFromFile($path)
+	{
+		$tokens = token_get_all(JFile::read($path));
```

Comment:
I'm also not sure on the restriction of `JFormField` as the base class name considering the API supports (but next to nobody uses, I think the stats plugin is the only core use of this feature) custom class prefixes.